### PR TITLE
Adding IterableAsDataPipe IterDataPipe

### PR DESCRIPTION
**Pull request by VitalyFedyunin**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63523 Adding DataLoader2 class as future replacement of DataLoader
* **#63522 Adding IterableAsDataPipe IterDataPipe**
Supports sharding and batching on loader level
* **#63522 Adding IterableAsDataPipe IterDataPipe
usefull for tests and simple cases**

usefull for tests and simple cases

Differential Revision: [D30426528](https://our.internmc.facebook.com/intern/diff/D30426528)

### CODE REVIEW

**Comment by VitalyFedyunin** [comment]
I had to copy it from torchdata as we need it for new DataLoader and test cases

**Review by ejguan — APPROVED** [review]
LGTM

File: torch/utils/data/datapipes/iter/utils.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from torch.utils.data import IterDataPipe
+
+
+class IterableAsDataPipeIterDataPipe(IterDataPipe):
+    def __init__(self, iterable):
+        self.iterable = iterable
+
+    def __iter__(self):
+        for data in self.iterable:
+            yield data
```
Comment (ejguan) [inline torch/utils/data/datapipes/iter/utils.py:10]: nit: `__len__` method

**Comment by VitalyFedyunin** [comment]
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30426528).

**Pull request closed without merging**